code/game/gamemodes/changeling/implements/powers/body.dm

```diff
@@ -249,6 +261,80 @@
 	to_chat(C, "<span class='notice'>We have regenerated fully.</span>")
 	remove_verb(C, /mob/proc/changeling_revive)
 
+/// Rip the changeling's head off as a last ditch effort to revive
+/mob/proc/changeling_emergency_transform()
+	set category = "Changeling"
+	set name = "Emergency Transform (1)"
+
+	var/datum/changeling/changeling = changeling_power(1, 0, 100, DEAD)
+	if(!changeling)
+		return
+
+	var/mob/living/carbon/human/H = src
+	if(!isturf(loc)) // so people can't transform inside places they should not, like sleepers
+		return
+
+	var/obj/item/organ/external/head = H.get_organ(BP_HEAD)
+	if(!head)
+		return
+
+	var/list/names = list()
+	for(var/datum/absorbed_dna/DNA in changeling.absorbed_dna)
+		names += "[DNA.name]"
+
+	var/S = input("Select the target DNA to reconfigure in our lesser form: ", "Target DNA", null) as null|anything in names
+	if(!S)
+		return
```

Comment:
I don't like this input - it makes this ability really hard to use in a pinch. It should just autoselect your current DNA.